Homework7/Calculator/Calculator/Calculator.cs

```diff
@@ -0,0 +1,264 @@
+                        CalculateAndPrint();
+                    }
+                    catch (Exception ex) when (ex is DivideByZeroException ||
+                                               ex is ArgumentException)
+                    {
+                        Clear();
+                        Display = ErrorMessage;
+                        return;
+                    }
+
+                    temporaryResult = resultOfCalculating;
+                    currentOperation = operation;
+                    currentState = States.IsOperand;
+                }
+                break;
+
+            case States.IsOperation:
+                currentOperation = operation;
+                HistoryDisplay = temporaryResult + " " + operation;
+                break;
+        }
+    }
+
+    private double Evaluate()
+        => currentOperation switch
+        {
+            "+/-" => resultOfCalculating == 0 ? 0 : -resultOfCalculating,
+            "+" => temporaryResult + resultOfCalculating,
+            "-" => temporaryResult - resultOfCalculating,
+            "×" => temporaryResult * resultOfCalculating,
+            "÷" => IsDoubleNumberEqualsToZero(resultOfCalculating)
+                    ? throw new DivideByZeroException("Division by zero is prohibited!")
+                    : temporaryResult / resultOfCalculating,
+            _ => throw new ArgumentException("Unknown operation!")
+        };
+
+    private bool IsDoubleNumberEqualsToZero(double number)
+    {
+        double delta = 0.001;
+        return Math.Abs(number) < delta;
+    }
+
+    /// <summary>
+    /// Calculates result number in the calculator.
+    /// </summary>
+    public void Calculate()
+    {
+        switch (currentState)
+        {
+            case States.IsOperand:
+                if (currentOperation != " ")
+                {
+                    try
+                    {
+                        CalculateAndPrint();
+                    }
+                    catch (Exception ex) when (ex is DivideByZeroException ||
+                                               ex is ArgumentException)
+                    {
+                        Clear();
+                        Display = ErrorMessage;
+                        return;
+                    }
+
+                    currentOperation = " ";
+                    currentState = States.IsOperand;
+                }
+                break;
+            case States.IsOperation:
+                try
+                {
+                    CalculateAndPrint();
+                }
+                catch (Exception ex) when (ex is DivideByZeroException ||
+                                           ex is ArgumentException)
+                {
+                    Clear();
+                    Display = ErrorMessage;
+                    return;
+                }
+                break;
+        }
+    }
+
+    private void CalculateAndPrint()
+    {
+        HistoryDisplay += " " + resultOfCalculating + " =";
```

Comment:
```suggestion
        HistoryDisplay = $"{HistoryDisplay} {resultOfCalculating} =";
```
Если нужно конкатенировать несколько строк, используй форматную строку.
Так и читаемее, и эффективнее (несколько последовательных конкатенаций порождают промежуточные строки.
Может, компилятор это соптимизирует, а может и нет.